### ISSUE TITLE
Accessibility support in microsoft wslg

### ISSUE THREAD

**Issue by carellagermano** [microsoft/wslg#580]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Hi,
I'm Germano Carella, a visually empaired developer from Italy.
I work on accessibility of aplications since 2002, and I discovered today wslg for windows.
I know how accessibility works on windows, but unfortunately I don't know how it works in X server.
I suspect that applications write directly on screen memory.
On linux there is only a solution, named orca, that has some accessiblity support for gnome desktop.
Infact, I tried to launch some x application, such as xcalc, it starts, but it isn't accessible for my screen reader.
I tried to get the window class of xcalc, and I noticed that is "RailsWindow"; so, I think is impossible to retrieve info directly from windows screen reading software.
So, I tried to install orca and run it.
When Orca starts, it returns an error: FileNotFoundError: no such file or directory 'xkbcomp'.
Orca is written in python language, so I cold study his code, but I need to start it and understand what is able to do with x apps.
Any idea?
Germano

**Comment by madewokherd** [comment]
Hi,

I don't know much about how WSLg works, but I've worked with accessibility on Linux.

I did a quick test of xcalc on Linux, and it's not accessible there either. You can use Accerciser to check. You'll probably need an application written in a toolkit like Qt or GTK.

Unfortunately, I think even if Orca works, its functionality will be limited on Wayland, which WSLg uses. See https://wiki.gnome.org/Accessibility/Wayland (not sure how up-to-date that page is)

I'm not familiar with xkbcomp, but it looks like a program you may need to install. On my Linux machine, it's in the x11-xkb-utils package.

I think we'd ultimately get the best result from "native" Windows accessibility support, but it's a lot of work. Accessibility on Linux is based on AT-SPI2, a DBus-based protocol. To connect this to MSAA/UIA, we'd need some sort of bridge running on Windows, and some way to access the accessibility DBus from Windows. We'd also need to correlate the Win32 windows with AT-SPI2 objects somehow.
